include/libdnf/rpm/nevra.hpp

```diff
@@ -149,6 +149,39 @@ inline void Nevra::set_arch(std::string && arch) {
 }
 
 
+/// Create a full nevra string (always contains epoch) from an object
+template <typename T>
+inline std::string to_full_nevra_string(const T & obj) {
+    auto epoch = obj.get_epoch();
+    if (epoch.empty()) {
+        epoch = "0";
+    }
+    return obj.get_name() + "-" + epoch + ":" + obj.get_version() + "-" + obj.get_release() + "." + obj.get_arch();
```

Comment:
What about to use std::string::reserce() and append? First calculate  required size, then reserve memory, then append() strings?

Reply:
That sounds like over-engineering to me, I prefer to keep the code simple.
I wanted to use std::ostringstream, but it was way slower.
I ended up with this code and @jrohel seemed to be fine with it when we discussed it over chat.

Reply:
Ok, changed that according to your suggestion. The performance increase is up to 25%.
This is separated to https://github.com/rpm-software-management/libdnf/pull/1042